build/tools/Automation/DotNet/Models/DotNetVersion.cs

```diff
@@ -35,8 +36,8 @@ public int CompareTo(object obj)
                 return 1;
             }
 
-            var thisVersion = new Version(this.Version);
-            var objVersion = new Version(objDotNetVersion.Version);
+            var thisVersion = SemanticVersion.Parse(this.Version);
+            var objVersion = SemanticVersion.Parse(objDotNetVersion.Version);
```

Comment:
Is there any validation with `SemanticVersion.Parse` that may bite us here that we wouldn't have seen previously with the `Version` constructor? For example, the `Version` constructor may just take the version string provided to it and attempt to piece together components of it to the properties of the `Version` class, whereas `SemanticVersion.Parse` will try to actually verify that each component aligns with a given property of the `SemanticVersion` class.

Along a similar path, below in the `VersionService` class, we're making a call to `SemanticVersion.TryParse`, which will swallow any exceptions possibly thrown when parsing a version string, so I'm wondering the order in which we'd be calling `TryParse` vs. `Parse` here.

Reply:
I think this should be ok here. `Version` will throw exceptions when it encounters releases, and `SemanticVersion.Parse` will do the same thing here. I think it should be safe since `SemVer.Parse` takes what `Version` could take and takes prereleases in addition. And if there's some issue, the behavior will be similar to what `Version` would be doing.

Reply:
OK sounds good -- it may be worth adding some validation in the future around these versions, but I think we should be fine in any case since we're pulling the versions from .NET (and other public sources) and not some random strings.